### ISSUE TITLE
Generic components | Generic type unresolved inside <script>

### ISSUE THREAD

**Issue by localusercamp** [nuxt/nuxt#21041]
### Environment

OS: Windows 10
Nuxt: 3.5.1
IDE: VSCode

### Reproduction

https://stackblitz.com/edit/github-sxsjje?file=app.vue

### Describe the bug

TS marks following code as error - unresolved.

```vue
<script setup lang="ts" generic="T">
defineProps<{
  data: T[] // unresolved
}>();
</script>
```

![image](https://github.com/nuxt/nuxt/assets/45385480/0bd8555e-f9de-4932-80ce-2336dded8d55)

### Additional context

Latest Volar installed
Latest Volar TS plugin installed
@builtin TS disabled

**Comment by localusercamp** [comment]
To solve this i tried to reinstall all tools.

Latest VSCode with only this extensions Installed:
![image](https://github.com/nuxt/nuxt/assets/45385480/aaf63ec4-56a2-4453-bc4d-3456bf74cb07)

Takeover mode enabled:
![image](https://github.com/nuxt/nuxt/assets/45385480/7a095774-8842-4fa0-a7eb-9cb4ee614bb8)
![image](https://github.com/nuxt/nuxt/assets/45385480/7cc25a1f-480f-4197-bb91-5c5c862301f6)

Fresh nuxt project created through `npx nuxi init`

But problem still occures
![image](https://github.com/nuxt/nuxt/assets/45385480/92c42931-6aa6-4e7a-a377-4485fca07291)

**Comment by Anoromi** [comment]
Having the same issue. **Solved it by switching to version 1.6.4 of Volar.** Also, pre-release version seems to work.
It seems there is some kind of bug where when specifying the latest version 1.6.5 in vscode it installs 1.2.0 instead. No idea what it causes, uninstalling and deleting extention didn't work.

**Comment by localusercamp** [comment]
> Having the same issue. **Solved it by switching to version 1.6.4 of Volar.** Also, pre-release version seems to work.
> It looks like there is some kind of bug where when specifying the latest version 1.6.5 in vscode it installs 1.2.0 instead. No idea what it causes, uninstalling and deleting extention didn't work. 

Did you switch versions for both volar and volar TS plugin, or only for volar?

**Comment by localusercamp** [comment]
> @localusercamp Only volar

Madness... Thank you a lot!
All works for me now.
The problem in VSCode extensions really.
What happening:
When you installing `TypeScript Vue Plugin (Volar)` vscode installing latest 1.6.5 version
But when you installing `Vue Language Features (Volar)` vscode installing the 1.2.0 version instead of real latest - 1.6.5 (same versioning as plugin).
I also cant switch to 1.6.5 - vscode installs 1.2.0 instead.
[Installing `Vue Language Features (Volar)` 1.6.4](https://stackoverflow.com/questions/42626065/vs-code-how-to-rollback-extension-install-specific-extension-version) do the trick.

Closing this, thanks @danielroe and @Anoromi.

P.S. My fault is that I did not check which version is installed, but trusted vscode instead, sorry 😓
